sklearn/utils/tests/test_validation.py

```diff
@@ -1050,7 +1050,8 @@ def __init__(self):
 def test_check_scalar_valid(x):
     """Test that check_scalar returns no error/warning if valid inputs are
     provided"""
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
```

Comment:
For future reviewers, I think the purpose of this test was to catch all warnings.